### ISSUE TITLE
lsp: support detecting test built on top of `Deno.test()` like std/testing/bdd

### ISSUE THREAD

**Issue by glebbash** [denoland/deno#14804]
**Is your feature request related to a problem? Please describe.**

Testing UI integration does not show tests defined written in BDD style (https://deno.land/std@0.136.0/testing/bdd.ts).

**Describe the solution you'd like**

Would be nice to have them displayed. For now the only workaround is to just use default Deno.test

**Comment by kitsonk** [comment]
This is currently a design limitation, as it would require a reliable way to statically analyze a module for tests without actually running the module.  Currently the static analysis is limited to a single module and identifying the `Deno.test` symbol and tracking tests that are registered.

Things like dynamically generated tests (or test steps) as well as other libraries that build on `Deno.test` cannot currently be analyzed and I personally currently can envision a good way to accomplish this at the moment. @dsherret I don't know if you have any thoughts on the topic.

**Comment by glebbash** [comment]
Definitely not an easy feature to implement but maybe it makes sense to implement it for `bdd` library only as it is part of std.

**Comment by KnorpelSenf** [comment]
I agree that static analysis for this would be very hard to do, and it is even harder to do it right. Doing it for `/std` only is the much more practical solution, albeit not as friendly towards third-party testing frameworks. However, I would argue that those frameworks are not equal to the standard library testing because, well, they're not in the standard library, and hence it is fine to only support `/std` here.

**Comment by nayeemrmn** [comment]
Ref https://github.com/denoland/vscode_deno/issues/572, https://github.com/denoland/vscode_deno/issues/793

**Comment by anion155** [comment]
Any update here?

**Comment by adamgreg** [comment]
This would be very useful... I've implemented a little wrapper and it's painful losing the VS Code support to easily debug tests or view results.

Could this be made easier to implement by adding requirements that wrappers must meet to be detected? e.g. a symbol property that the wrapper function should include.

**Comment by yasaichi** [comment]
https://github.com/denoland/vscode_deno/issues/793 
seems related to this issue.
